File: stress_test.py
import requests
import time

urls = [
    "https://www.amazon.in/dp/B08V8W3K9K",
    "https://www.flipkart.com/apple-iphone-15-black-128-gb/p/itm6ac6485515ae4",
    "https://www.myntra.com/shoes/nike/nike-men-black-air-zoom-pegasus-38-running-shoes/14318112/buy",
    "https://www.meesho.com/s/p/4p6z8z",
    "https://www.ajio.com/s/p/461234567"
]

def stress_test():
    print("Starting Stress Test: Sequential Analysis...")
    
    for i, url in enumerate(urls):
        print(f"\n[{i + 1}/{len(urls)}] Analyzing: {url}")
        
        try:
            start = time.time()
            response = requests.post("http://localhost:8000/fetch-product-intelligence", json={"url": url})
            end = time.time()
            
            if response.status_code == 200:
                data = response.json()
                if data.get("success"):
                    print(f"[SUCCESS] {end - start:.2f}s")
                    print(f"Title: {data.get('title')[:50]}...")
                else:
                    print(f"[FAILURE] Backend reported failure: {data.get('error')}")
            else:
                print(f"[ERROR] HTTP Error {response.status_code}: {response.text}")
        except Exception as e:
            print(f"[CRASH] Request crashed: {e}")
    
    print("\nStress Test Complete.")

if __name__ == "__main__":
    stress_test()

File: stress_test.js
import axios from 'axios';

const urls = [
    "https://www.amazon.in/ASIAN-Powerplay-01-Running-Walking-Technology/dp/B08V8W3K9K",
    "https://www.amazon.in/Samsung-Galaxy-Ultra-Storage-Graphite/dp/B0BW8N3X6D",
    "https://www.amazon.in/Sony-WH-1000XM4-Bluetooth-Headphones-Cancellation/dp/B0863TXGM3"
];

async function stressTest() {
    console.log("Starting Stress Test: Sequential Analysis...");
    
    for (let i = 0; i < urls.length; i++) {
        const url = urls[i];
        console.log(`\n[${i + 1}/${urls.length}] Analyzing: ${url}`);
        
        try {
            const start = Date.now();
            const response = await axios.post("http://localhost:8000/fetch-product-intelligence", { url });
            const end = Date.now();
            
            if (response.data.success) {
                console.log(`✅ Success in ${((end - start) / 1000).toFixed(2)}s`);
                console.log(`Title: ${response.data.title.substring(0, 50)}...`);
            } else {
                console.log(`❌ Backend reported failure: ${response.data.error}`);
            }
        } catch (error) {
            console.log(`❌ Request crashed: ${error.message}`);
        }
    }
    
    console.log("\nStress Test Complete.");
}

stressTest();
